helix-term/src/commands/typed.rs

```diff
@@ -1952,6 +1952,34 @@ fn run_shell_command(
     Ok(())
 }
 
+fn clear_register(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+    ensure!(args.len() <= 1, ":clear-register takes at most 1 argument");
+    if args.is_empty() {
+        cx.editor.registers.clear();
+        cx.editor.set_status("All registers cleared");
+        return Ok(());
+    }
+
+    ensure!(args[0].len() == 1, format!("Invalid register {}", args[0]));
```

Comment:
Won't this cause problems with keyboards that have non ASCII keys? A register can be any character on the keyboard and `len()` only counts the number of bytes. `args[0].chars().count()` would be more appropriate in that case.

Reply:
Thanks for pointing that out, I pushed the corrected version.